tests/asgi/test_boundedstream_asgi.py

```diff
@@ -164,6 +164,30 @@ async def test_iteration():
     falcon.async_to_sync(test_iteration)
 
 
+@falcon.runs_sync
+async def test_iterate_streaming_request():
+    events = iter(
+        (
+            {'type': 'http.request', 'body': b'Hello, ', 'more_body': True},
+            {'type': 'http.request', 'body': b'World', 'more_body': True},
+            {'type': 'http.request', 'body': b'!\n', 'more_body': True},
+            {'type': 'http.request', 'body': b'', 'more_body': False},
+            {'type': 'http.disconnect'},
```

Comment:
Do we want to not break out of the loop before consuming the http.disconnect event? Hopefully ASGI servers don't care if the app ever receives the disconnect...

Reply:
We do want, and the code handles it correctly (there are many other tests verifying this behavior). Maybe it's not completely obvious what this means since it's an isolated unit test... But if iteration is used to parse the request body, one cannot wait for a disconnect from the client which is still waiting for a response (the very crux of this bug). I'll add a `NOTE` in a follow up PR.